client/src/components/LoginSession/LoginSessionContainer.js

```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types';
+// import React, { Fragment } from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import backend from 'lib/Backend';
+import Config from 'lib/Config'; // eslint-disable-line
+import LoginSession from './LoginSession';
+import { success, error } from 'state/toasts/ToastsActions';
+
+// Handle communication with  server on logout and toast notifications via redux
+// Using a class component rather than function component so that LoginSessionContainer-test.js
+// can use wrapper.instance() to test logout()
+class LoginSessionContainer extends Component {
+    //
+    constructor(props) {
+        super(props);
+        this.createEndpoint = this.createEndpoint.bind(this);
+        this.logout = this.logout.bind(this);
+    }
+
+    createEndpoint() {
+        return backend.createEndpointFetcher({
+            url: `${this.props.LogOutEndpoint}/:id`,
+            method: 'delete',
+            payloadSchema: {
+                id: { urlReplacement: ':id', remove: true },
+                SecurityID: { querystring: true }
+            }
+        });
+    }
+
+    // params are 3x callbacks defined within LoginSession
+    logout(beforeRequest, receivedResponse, caughtException) {
+        beforeRequest();
+        const endpoint = this.createEndpoint();
+        endpoint({
+            id: this.props.ID,
+            SecurityID: Config.get('SecurityID')
+        })
+        .then(response => {
+            const failed = !response.success;
+            receivedResponse(failed);
+            if (failed) {
+                this.props.displayToastFailure(response.message);
+            } else {
+                this.props.displayToastSuccess(response.message);
+            }
+        })
+        .catch(() => {
+            caughtException();
+        });
+    }
+
+    render() {
+        const { ID, LogoutEndPoint, ...loginSessionProps } = { ...this.props };
```

Comment:
You're destructuring `this.props`, then restructuring it, only to destructure it again into the variable you need.

```suggestion
        const { ID, LogoutEndPoint, ...loginSessionProps } = this.props;
```

Reply:
Updated